### ISSUE TITLE
Support for AsyncIO

### ISSUE THREAD

**Issue by siddhantgoel** [siddhantgoel/streaming-form-data#41]
Building `async` support into this package would make a lot of sense. One use case, for example, would be to do networked IO in parallel while file chunks are being uploaded. This is required for [S3 uploads](https://github.com/siddhantgoel/streaming-form-data/issues/3) (or basically uploads to any other file hosting service).

The API for how this should look like is not yet clear. Async code (at least how Python implements it) tends to "split" the codebase into sync and async parts, so we would need to figure that one out first.

There's a discussion and some ideas in https://github.com/siddhantgoel/streaming-form-data/pull/29 .

**Comment by mehmetemreakbulut** [comment]
Is there any plan to build async support? Currently I am using FastApi and UploadFile class to get the file input. It writes the data to disk, so not efficient when large zip files are uploaded.

**Comment by siddhantgoel** [comment]
Yes, I'd really like to have async support built into this library but at the moment have time constraints on my end.
